### PR TITLE
Let the user choose the language with a select-box instead of typing the language code into a input-field.

### PR DESCRIPTION
**Pull request by gonzoMD**
* Add langcodes.php which contains an array that holds the ISO 639-1 data
* Use a select-box in index.php, encoding.php and diff.php instead of an input-field
* Change some texts accordingly

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -14,11 +15,16 @@
...
     <form method="GET" class="form-horizontal">
         <fieldset>
-            <legend>Please type your <a href="https://beta.wikiversity.org/wiki/List_of_ISO_639-1_codes">language code in ISO 639-1</a>. For example: pl for Polish, de for German</legend>
+            <legend>Please choose your language from the list below. </legend>
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control">
```
Comment (Saibamen) [inline index.php:22]: Please add removed `required="required"`

File: index.php [hunk 2]
```diff
@@ -14,11 +15,16 @@
...
+                        <option value="" selected disabled hidden>Choose here</option>
```
Comment (Saibamen) [inline index.php:23]: Delete `selected` and use it in `foreach` loop - depending on `$_SESSION['lang']`

File: index.php [hunk 3]
```diff
@@ -14,11 +15,16 @@
...
+                        <?php foreach ($langcodes as $language) {
+    echo"<option value='$language[0]'>$language[1]</option>";
+}?>
```
Comment (Saibamen) [inline index.php:26]: Fix identation

File: langcodes.php [hunk 4]
```diff
@@ -0,0 +1,195 @@
+<?php
+/* PROJECT:     ReactOS Translation Tool
+ * LICENSE:     GPL
+ * AUTHORS:     Robert Naumann <gonzomdx[at]gmail.com>
+ */
+
+// Data taken from: https://beta.wikiversity.org/wiki/List_of_ISO_639-1_codes
```
Comment (Saibamen) [inline langcodes.php:7]: Please add additional comment like `// Last updated: 13.04.2021`

**Review by Saibamen — CHANGES_REQUESTED** [review]

**Comment by gonzoMD** [comment]
all done. but after fixing the indent styleci goes nuts.

File: diff.php [hunk 5]
```diff
@@ -14,11 +15,17 @@
...
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo"<option value='$language[0]'";
+                        if(isset($_SESSION['lang'])&& ($language[0]== $_SESSION['lang'])) echo "selected";
+                        echo"> $language[1]</option>";
```
Comment (Saibamen) [inline diff.php:26]:
```suggestion
                        echo '<option value="'. $language[0] .'" ';
                        if (isset($_SESSION['lang']) && ($language[0] == $_SESSION['lang'])) echo 'selected';
                        echo '> $language[1]</option>';
```

File: encoding.php [hunk 6]
```diff
@@ -14,11 +15,17 @@
...
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo"<option value='$language[0]'";
+                        if(isset($_SESSION['lang'])&& ($language[0]== $_SESSION['lang'])) echo "selected";
+                        echo"> $language[1]</option>";
```
Comment (Saibamen) [inline encoding.php:26]:
```suggestion
                        echo '<option value="'. $language[0] .'" ';
                        if (isset($_SESSION['lang']) && ($language[0] == $_SESSION['lang'])) echo 'selected';
                        echo '> $language[1]</option>';
```

File: index.php [hunk 7]
```diff
@@ -14,11 +15,17 @@
...
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo"<option value='$language[0]'";
+                        if(isset($_SESSION['lang'])&& ($language[0]== $_SESSION['lang'])) echo "selected";
+                        echo"> $language[1]</option>";
```
Comment (Saibamen) [inline index.php:26]:
```suggestion
                        echo '<option value="'. $language[0] .'" ';
                        if (isset($_SESSION['lang']) && ($language[0] == $_SESSION['lang'])) echo 'selected';
                        echo '> $language[1]</option>';
```

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: diff.php [hunk 8]
```diff
@@ -14,11 +15,17 @@
...
-            <legend>Please type your <a href="https://beta.wikiversity.org/wiki/List_of_ISO_639-1_codes">language code in ISO 639-1</a>. For example: pl for Polish, de for German</legend>
+            <legend>Please choose your language and the directories, where you want to search for untranslated strings, from the lists below.</legend>
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo '<option value="'. $language[0] .'" ';
```
Comment (Saibamen) [inline diff.php:24]:
```suggestion
                        echo '<option value="'.$language[0].'" ';
```

File: encoding.php [hunk 9]
```diff
@@ -14,11 +15,17 @@
...
-            <legend>Please type your <a href="https://beta.wikiversity.org/wiki/List_of_ISO_639-1_codes">language code in ISO 639-1</a>. For example: pl for Polish, de for German</legend>
+            <legend>Please choose your language and the directories, where you want to search for untranslated strings, from the lists below.</legend>
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo '<option value="'. $language[0] .'" ';
```
Comment (Saibamen) [inline encoding.php:24]:
```suggestion
                        echo '<option value="'.$language[0].'" ';
```

File: index.php [hunk 10]
```diff
@@ -14,11 +15,17 @@
...
-            <legend>Please type your <a href="https://beta.wikiversity.org/wiki/List_of_ISO_639-1_codes">language code in ISO 639-1</a>. For example: pl for Polish, de for German</legend>
+            <legend>Please choose your language from the list below. </legend>
             <div class="form-group">
-                <label class="col-md-4 control-label" for="lang">Language code:</label>
+                <label class="col-md-4 control-label" for="lang">Language:</label>
                 <div class="col-md-4">
-                    <input type="text" value="<?php echo isset($_SESSION['lang']) ? $_SESSION['lang'] : '' ?>" id="lang" name="lang" class="form-control input-md" required="required" autofocus="autofocus" pattern="[A-Za-z]{2}" title="Two letter language code"/>
+                    <select id="lang" name="lang" class="form-control" required="required">
+                        <?php foreach ($langcodes as $language) {
+                        echo '<option value="'. $language[0] .'" ';
```
Comment (Saibamen) [inline index.php:24]:
```suggestion
                        echo '<option value="'.$language[0].'" ';
```

**Comment by Saibamen** [comment]
Merged. Thanks

**Pull request merged**